lib/utils/types.ts

```diff
@@ -0,0 +1,50 @@
+import { Address, Hex } from "viem";
+
+/**
+ * @notice Struct used to define the domain of the typed data signature, defined in EIP-712.
+ * @param name The user friendly name of the contract that the signature corresponds to.
+ * @param version The version of domain object being used.
+ * @param chainId The ID of the chain that the signature corresponds to (ie Ethereum mainnet: 1, Goerli testnet: 5, ...).
+ * @param verifyingContract The address of the contract that the signature pertains to.
+ */
+export type DomainData = {
+  name: string;
+  version: string;
+  chainId: number;
+  verifyingContract: `0x${string}`;
+};
+
+/**
+ * @notice Struct used to define a parameter for off-chain Database Handler deferral.
+ * @param name The variable name of the parameter.
+ * @param value The string encoded value representation of the parameter.
+ */
+export type Parameter = {
+  name: string;
+  value: string;
+};
+
+/**
+ * @notice Struct used to define the message context used to construct a typed data signature, defined in EIP-712,
+ * to authorize and define the deferred mutation being performed.
+ * @param functionSelector The function selector of the corresponding mutation.
+ * @param sender The address of the user performing the mutation (msg.sender).
+ * @param parameter[] A list of <key, value> pairs defining the inputs used to perform the deferred mutation.
+ */
+export type MessageData = {
+  functionSelector: `0x${string}`;
+  sender: `0x${string}`;
```

Comment:
@lgahdl  we have type address from viem

Reply:
okay, we can replace this by the viem's Address, but I didn't add this MessageData type in this task, I only moved all types from blockchain-txs.ts to a new file to keep the types in a separated file.

Reply:
Nice! created the issue: https://github.com/blockful-io/external-resolver-dapp/issues/126